CHANGES

```diff
@@ -9,6 +9,32 @@
 
  Changes between 1.1.1 and 3.0.0 [xx XXX xxxx]
 
+  *) All of the low level DSA functions have been deprecated including:
+
+     DSA_do_sign, DSA_do_verify, DSA_OpenSSL, DSA_set_default_method,
+     DSA_get_default_method, DSA_set_method, DSA_get_method, DSA_new,
+     DSA_new_method, DSA_free, DSA_up_ref, DSA_size, DSA_bits,
```

Comment:
We still need to be able to create DSA objects/free them/up-ref them - so I think this goes a step too far. Applications should be able to get/set the pqg params (and I note you don't deprecate those) and then assign the DSA objects to an EVP_PKEY.

Reply:
Do we want to be able to do this from outside libcrypto?  I thought the idea was to make the DSA structures internal.

Reply:
Well, I currently see no reason to prevent people from creating a DSA object if all they're going to do with it is assign it an EVP_PKEY. Its calling the low level *crypto* routines that we want to prevent.

Reply:
I think I've undeprecated all of the relevant calls.
I also left the p, q, g, public and private key getters and setters.